src/cmd-run

```diff
@@ -173,7 +203,26 @@ EOF
         exit 1
     fi
 fi
-set -- -fw_cfg name=opt/com.coreos/config,file="${IGNITION_CONFIG_FILE}" "$@"
+
+if [ "${VM_INJECT}" = 1 ]; then
+    export LIBGUESTFS_BACKEND=direct
+    eval "$(guestfish --listen)"
+    guestfish --remote -- add ${VM_DISK}
+    guestfish --remote -- run
+    BOOT=$(guestfish --remote -- findfs-label boot)
+    guestfish --remote -- mount "${BOOT}" /
+    mkdir -p tmp/ignition
+    cat ${IGNITION_CONFIG_FILE} > tmp/ignition/config.ign
```

Comment:
Why  `cat` instead of e.g. `cp --reflink=auto "${IGNITION_CONFIG_FILE} tmp/ignition/config.ign`?

Reply:
Another concern I have here is around SELinux labeling; any files we create via libguestfs need to be labeled correctly.  This *might* work today accidentally?  Regardless we'll likely need to beef up our support for that in our internal libguestfs helper.

Reply:
@cgwalters what should be the correct selinux label for the ignition file?

Reply:
That is...a tricky topic.  I'll take a look at adding a primitive to our internal `libguestfs.sh` for this - see related code in https://github.com/cgwalters/qcow2-to-vagrant/blob/master/qcow2-to-vagrant

Reply:
OK I tossed up https://github.com/coreos/coreos-assembler/pull/444 but I am having other apparent troubles with my local build after the Ignition v3 switch, haven't debugged that yet.

Reply:
And here's my work so far on top of this PR:

```
From 8a48c8ca33c2ac34f43e8a15b3fcd148f052d028 Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Thu, 28 Mar 2019 00:24:56 +0000
Subject: [PATCH] wip

---
 src/cmd-run | 17 ++++++-----------
 1 file changed, 6 insertions(+), 11 deletions(-)

diff --git a/src/cmd-run b/src/cmd-run
index d629b41c..3dc59212 100755
--- a/src/cmd-run
+++ b/src/cmd-run
@@ -10,6 +10,8 @@ set -euo pipefail
 dn=$(dirname "$0")
 # shellcheck source=src/cmdlib.sh
 . "${dn}"/cmdlib.sh
+# shellcheck source=src/libguestfish.sh
+. "${dn}"/libguestfish.sh
 
 VM_DISK=
 VM_MEMORY=2048
@@ -190,17 +192,10 @@ EOF
 fi
 
 if [ "${VM_INJECT}" = 1 ]; then
-    export LIBGUESTFS_BACKEND=direct
-    eval "$(guestfish --listen)"
-    guestfish --remote -- add ${VM_DISK}
-    guestfish --remote -- run
-    BOOT=$(guestfish --remote -- findfs-label boot)
-    guestfish --remote -- mount "${BOOT}" /
-    mkdir -p tmp/ignition
-    cat ${IGNITION_CONFIG_FILE} > tmp/ignition/config.ign
-    guestfish --remote -- copy-in "$(realpath tmp/ignition)" /
-    guestfish --remote -- exit
-    rm -rf tmp/ignition
+    coreos_gf_run_mount "${VM_DISK}"
+    coreos_gf upload ${IGNITION_CONFIG_FILE} /boot/config.ign
+    coreos_gf_relabel /boot/config.ign
+    coreos_gf_shutdown
 fi
 
 case $arch in
-- 
2.20.1

```

Reply:
Yup, mine biggest issue atm is the after forward porting my changes seem to not work anymore..., might be the selinux but yeah I need to dig deeper again...

Reply:
Forwardported and some fix ups and it works again for me